### ISSUE TITLE
HiddenField.ignoreSetter does not work for single-letter fields since 6.3

### ISSUE THREAD

**Issue by Derbeth** [checkstyle/checkstyle#730]
Given Java source:

```
package com.example;

public class Point {

    private int x;

    private int yy;

    public void setX(int x) {
        this.x = x;
    }

    public void setYy(int yy) {
        this.yy = yy;
    }
}
```

and Checkstyle configuration:

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="HiddenField">
            <property name="ignoreSetter" value="true"/>
        </module>
    </module>
</module>
```

Checkstyle reports the following error:

    Point.java:9:26: 'x' hides a field.

The regression appears in Checkstyle 6.3, 6.4 and 6.4.1. Works fine for 6.2, 6.1, 6.0 and 5.9.

[Release notes for 6.3](http://checkstyle.sourceforge.net/releasenotes.html#Release_6.3) mention HiddenField in new implemented features, but the provided link to #589 points to a completely unrelated commit.

**Comment by mkordas** [comment]
Issue confirmed, thanks for submitting. Indeed this bug was introduced in commit 6784e5bcd5baa6b5086bf8b63709856d685a74c8. Please compare how `capitalize()` works [before and after changes](https://github.com/checkstyle/checkstyle/commit/6784e5bcd5baa6b5086bf8b63709856d685a74c8#diff-bd2095a110f21178164b052b84a3eb32R373) for one letter string.

I'm working to fix that.

**Comment by romani** [comment]
merged , will be in 6.5
